components/MainLayout.js

```diff
@@ -10,9 +11,35 @@ const ga = `
   ga('send', 'pageview');
 `
 
+const isKeyPressedWithClick = (event) => event.ctrlKey || event.altKey || event.shiftKey || event.metaKey
+const isExternal = (targetUrl) => {
+  const stripHash = (url) => url.replace(/#.*$/, '')
+  return stripHash(targetUrl) !== stripHash(window.location.href)
+}
+
+const handleLinkClick = (event) => {
+  let target = event.target
+  while (target && target.nodeName !== 'A') {
+    target = target.parentNode
+  }
+
+  if (target && !isExternal(target.href) && isKeyPressedWithClick(event)) {
+    event.preventDefault()
```

Comment:
Right now, Cmd+Clicking on an internal link doesn’t do anything (it should open the link in a new tab).

To fix this, this line should be changed to say `return`, which will make the browser perform the default behavior (opening the link in a new tab).

However, the code is still complicated. Therefore, I recommend changing lines 26~28 to a series of early-returns.

```js
if (!target) return
if (isExternal(target.href)) return
if (isKeyPressedWithClick(event)) return
```

Then, you can remove the `if` clause in line 30 and 37, because all these conditions are already met:

1. User clicked on a link
2. The link is internal
3. Link is clicked without any modifier key